c10/core/TensorImpl.cpp

```diff
@@ -118,17 +118,7 @@ bool TensorImpl::is_contiguous(at::MemoryFormat memory_format) const {
   AT_ASSERT(compute_contiguous() == is_contiguous_);
 #endif
   if (memory_format == at::MemoryFormat::ChannelsLast) {
-    if (dim() == 4) {
-      auto strides_1 = 1;
-      auto strides_3 = sizes_[1];
-      auto strides_2 = strides_3 * sizes_[3];
-      auto strides_0 = strides_2 * sizes_[2];
-      if (strides_0 == strides_[0] && strides_1 == strides_[1] &&
-          strides_2 == strides_[2] && strides_3 == strides_[3]) {
-        return true;
-      }
-    }
-    return false;
+      return dim() == 4 && strides() == get_channels_last_strides(sizes());
```

Comment:
is C++ smart enough to inline std::vector and avoid memory allocation? if not, I wouldn't pull out this function

Reply:
This function call will go away in the next PR (Where I introduce 'tag') as calculating this every time is very inefficient.

Reply:
Is the tag advisory or not? If the tag is advisory, I would not expect `is_contiguous` to change behavior depending on if the tag is set or not (and so I agree with @dzhulgakov here). If the tag is non-advisory, I have concerns about the overall design :)

Reply:
Since we killed tags, now this comment applies once again, right?

Reply:
This code line will go away anyway in non 'tag' option too.

Reply:
For peace of mind, I will not pull out it now. And replace copy-pasted code in the next PR.